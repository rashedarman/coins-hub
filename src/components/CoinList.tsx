import { SimpleGrid } from '@mantine/core';
import { Coins } from '../interfaces';
import CoinItem from './CoinItem';
import PageWrapper from './PageWrapper';

function CoinsList(props: Coins) {
  const { coins } = props;

  return (
    <PageWrapper>
      <SimpleGrid
        cols={4}
        breakpoints={[
          { maxWidth: 'md', cols: 3 },
          { maxWidth: 'sm', cols: 2 },
          { maxWidth: 'xs', cols: 1 },
        ]}
        my="xl"
      >
        {coins.map((coin) => (
          <CoinItem key={coin.id} coin={coin} />
        ))}
      </SimpleGrid>
    </PageWrapper>
  );
}

export default CoinsList;
