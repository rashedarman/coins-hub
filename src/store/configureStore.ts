import { configureStore } from '@reduxjs/toolkit';
import cryptoSlice from './crypto/cryptoSlice';
import newsSlice from './news/newsSlice';

const rootReducer = {
  crypto: cryptoSlice,
  news: newsSlice,
};

const store = configureStore({
  reducer: rootReducer,
});

export default store;
