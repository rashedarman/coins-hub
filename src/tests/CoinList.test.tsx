import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import CoinList from '../components/CoinList';
import store from '../store/configureStore';
import { coins } from './__mockData';

describe('Testing CoinsListTable component', () => {
  it('Renders CoinsListTable component correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <BrowserRouter>
          <CoinList coins={coins} />
        </BrowserRouter>
      </Provider>,
    );
    expect(container).toMatchSnapshot();
  });
});
