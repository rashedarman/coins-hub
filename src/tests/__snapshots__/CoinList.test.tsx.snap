// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CoinsListTable component Renders CoinsListTable component correctly 1`] = `
<div>
  <div
    class="mantine-Container-root mantine-sz575t"
  >
    <div
      class="mantine-SimpleGrid-root mantine-1xwjr2u"
    >
      <div
        class="mantine-Paper-root mantine-Card-root mantine-1swzeoq"
      >
        <div
          class="mantine-73o8aw"
        >
          <div
            class="mantine-Image-root mantine-15po0m8"
            style="width: 60px;"
          >
            <figure
              class="mantine-11nhzn5 mantine-Image-figure"
            >
              <div
                class="mantine-qqmv3w mantine-Image-imageWrapper"
              >
                <img
                  class="mantine-3y8yz3 mantine-Image-image"
                  src="https://static.coinstats.app/coins/1650455588819.png"
                  style="object-fit: cover; width: 60px; height: auto;"
                />
              </div>
            </figure>
          </div>
          <div
            class="mantine-Badge-root mantine-6iang"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              RANK #
              1
            </span>
          </div>
        </div>
        <div
          class="mantine-Text-root mantine-11o1k2d"
        >
          Bitcoin
           
          (
          BTC
          )
        </div>
        <div
          class="mantine-879la3"
        >
          <div
            class="mantine-Text-root mantine-1l0xknd"
          >
            $22,959.00
          </div>
          <div>
            <div
              class="mantine-Badge-root mantine-1w333o4"
            >
              <span
                class="mantine-h9iq4m mantine-Badge-inner"
              >
                ▼
                -0.15
                %
              </span>
            </div>
          </div>
        </div>
      </div>
      <div
        class="mantine-Paper-root mantine-Card-root mantine-1swzeoq"
      >
        <div
          class="mantine-73o8aw"
        >
          <div
            class="mantine-Image-root mantine-15po0m8"
            style="width: 60px;"
          >
            <figure
              class="mantine-11nhzn5 mantine-Image-figure"
            >
              <div
                class="mantine-qqmv3w mantine-Image-imageWrapper"
              >
                <img
                  class="mantine-3y8yz3 mantine-Image-image"
                  src="https://static.coinstats.app/coins/1633034449328.png"
                  style="object-fit: cover; width: 60px; height: auto;"
                />
              </div>
            </figure>
          </div>
          <div
            class="mantine-Badge-root mantine-6iang"
          >
            <span
              class="mantine-h9iq4m mantine-Badge-inner"
            >
              RANK #
              2
            </span>
          </div>
        </div>
        <div
          class="mantine-Text-root mantine-11o1k2d"
        >
          Ethereum
           
          (
          ETH
          )
        </div>
        <div
          class="mantine-879la3"
        >
          <div
            class="mantine-Text-root mantine-1l0xknd"
          >
            $1,727.00
          </div>
          <div>
            <div
              class="mantine-Badge-root mantine-1w333o4"
            >
              <span
                class="mantine-h9iq4m mantine-Badge-inner"
              >
                ▼
                -0.24
                %
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
